server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -113,14 +121,18 @@ public final class IndicesRequestCache implements RemovalListener<IndicesRequest
         this.size = INDICES_CACHE_QUERY_SIZE.get(settings);
         this.expire = INDICES_CACHE_QUERY_EXPIRE.exists(settings) ? INDICES_CACHE_QUERY_EXPIRE.get(settings) : null;
         long sizeInBytes = size.getBytes();
-        CacheBuilder<Key, BytesReference> cacheBuilder = CacheBuilder.<Key, BytesReference>builder()
-            .setMaximumWeight(sizeInBytes)
-            .weigher((k, v) -> k.ramBytesUsed() + v.ramBytesUsed())
-            .removalListener(this);
-        if (expire != null) {
-            cacheBuilder.setExpireAfterAccess(expire);
-        }
-        cache = cacheBuilder.build();
+        StoreAwareCacheBuilder<Key, BytesReference> openSearchOnHeapCacheBuilder = new OpenSearchOnHeapCache.Builder<Key, BytesReference>()
+            .setWeigher((k, v) -> k.ramBytesUsed() + v.ramBytesUsed())
+            .setMaximumWeightInBytes(sizeInBytes)
+            .setExpireAfterAccess(expire);
+
+        cache = new TieredSpilloverCache.Builder<Key, BytesReference>().setOnHeapCacheBuilder(openSearchOnHeapCacheBuilder)
+            .setListenerConfiguration(
+                new StoreAwareCacheEventListenerConfiguration.Builder<Key, BytesReference>().setEventListener(this)
+                    .setEventTypes(EnumSet.of(EventType.ON_CACHED, EventType.ON_HIT, EventType.ON_MISS, EventType.ON_REMOVAL))
+                    .build()
+            )
+            .build();
     }
```

Comment:
I would recommend to do this integration related change post some perf testing. The perf test should show that after using `TieredSpilloverCache` with just onHeap cache in it, there is no regression in the perf and functionality.

Reply:
Will do. Removed it for now.